hbase-common/src/main/java/org/apache/hadoop/hbase/util/Threads.java

```diff
@@ -197,4 +199,33 @@ public static void setLoggingUncaughtExceptionHandler(Thread t) {
   public static void printThreadInfo(PrintStream stream, String title) {
     ReflectionUtils.printThreadInfo(stream, title);
   }
+
+  /**
+   * Checks whether any non-daemon thread is running.
+   * @return true if there are non daemon threads running, otherwise false
+   */
+  public static boolean isNonDaemonThreadRunning() {
+    AtomicInteger nonDaemonThreadCount = new AtomicInteger();
+    Set<Thread> threads =  Thread.getAllStackTraces().keySet();
+    threads.forEach(t -> {
+      // Exclude current thread
+      if (t.getId() != Thread.currentThread().getId() && !t.isDaemon()) {
+        nonDaemonThreadCount.getAndIncrement();
+        LOG.info("Non daemon thread {} is still alive", t.getName());
+        LOG.info(printStackTrace(t));
+      }
+    });
+    return nonDaemonThreadCount.get() > 0;
+  }
+
+  /*
+    Print stack trace of the passed thread
+   */
+  public static String printStackTrace(Thread t) {
+    StringBuilder sb = new StringBuilder();
+    for (StackTraceElement frame: t.getStackTrace()) {
+      sb.append("\n").append("    " + frame.toString());
```

Comment:
nit: `append("   ").append(frame.toString())`

Reply:
Kept multiple spaces so as to make it look like stack trace.

Reply:
Oh no not that, I meant adding `frame.toString()` with `.append()` rather than String concat. But it's minor nit only. Maybe you can change this and keep backport PRs in line with this, and all can be merged together.